File: library/src/main/java/com/github/ihsg/patternlocker/IIndicatorLinkedLineView.java
package com.github.ihsg.patternlocker;

import android.graphics.Canvas;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

import java.util.List;

/**
 * Created by hsg on 22/02/2018.
 */

public interface IIndicatorLinkedLineView {
    /**
     * 绘制指示器连接线
     *
     * @param canvas
     * @param hitList
     * @param cellBeanList
     * @param isError
     */
    void draw(@NonNull Canvas canvas,
              @Nullable List<Integer> hitList,
              @NonNull List<CellBean> cellBeanList,
              boolean isError);
}

File: library/src/main/java/com/github/ihsg/patternlocker/IIndicatorLinkedLineView.kt
package com.github.ihsg.patternlocker

import android.graphics.Canvas

/**
 * Created by hsg on 22/02/2018.
 */

interface IIndicatorLinkedLineView {
    /**
     * 绘制指示器连接线
     *
     * @param canvas
     * @param hitIndexList
     * @param cellBeanList
     * @param isError
     */
    fun draw(canvas: Canvas,
             hitIndexList: List<Int>,
             cellBeanList: List<CellBean>,
             isError: Boolean)
}